src/build123d/joints.py

```diff
@@ -87,28 +87,29 @@ def __init__(
         super().__init__(label, to_part)
 
     @overload
-    def connect_to(self, other: BallJoint, *, angles: RotationLike = None, **kwargs):
+    def connect_to(self, other: BallJoint, *args, angles: RotationLike = None, **kwargs):
```

Comment:
`*` is not equivalent to `*args`

Reply:
Thanks @MatthiasJ1. Can you say more about what problem you see here?
(I'm newly returning to python work so wouldn't be shocked if I'm missing something silly here)

Reply:
`*,` forces subsequent argument to be explicitly passed as keyword arguments. In this case it would make `a.connect_to(b, foo)` invalid and would require `a.connect_to(b, angles=foo)`. see https://peps.python.org/pep-3102/
That _may_ still be the right choice here but that's a matter of API design not simply cleanup refactoring.

Reply:
Thanks @snoyer. IIUC `a.connect_to(b, foo)` is invalid both in the "before" and "after" versions of the code being pointed at here, and `a.connect_to(b, angles=foo)` is valid in both. If that's true, then the change seems like a no-op. 
Just to verify my understanding I wrote https://colab.research.google.com/drive/1fAbilkOI2QNnYZ6-2JaQipCkxH9fJt_R which seems to show identical behavior for simplified versions of the before/after code. 
Do you have an example of a call that is valid before and invalid after, or invalid before (and where catching the invalidity is useful, I suppose) but silently "valid" after?

Reply:
good point, my example was poorly chosen. I guess the `*args,` version would accept unnamed (unused?) positional arguments that the `*, ` version wouldn't (eg. `g(1,2,b=3)` is ok but `f(1,2,b=3)` isn't).
Again, whether that's good/better or bad/worse is up to API considerations. Just splitting hairs about it not being a 100% equivalent refactor here :)

Reply:
Thanks again @snoyer. I definitely appreciate the double- and triple-checking here.
If the API change is judged ok by @gumyr I'm happy to reword the commit/PR descriptions to reflect this.
Will flip the PR to "draft" mode pending that judgement.

Reply:
I'd like the `*,` to remain thus forcing only kwargs.  Joints parameters are already kinda complex and I expect them to get more so over time so I want the clarity that comes with kwargs.

Reply:
AFAICT `*, **kwargs` in succession isn't valid Python, and an intervening (named) keyword parameter is required, so the best I've been able to come up with so far is https://github.com/gumyr/build123d/commit/85ee55fabd5b981894bb0d0cd1ca90f8c0dabeb1
@snoyer @MatthiasJ1: do you see any issues with that approach?
@gumyr Please LMK if that looks good to you and I'll rework this PR on top of that commit.

Reply:
I don't have time right now to properly investigate but I don't like:
```py
def connect_to(self, other: Joint, *, ignored: int = 0, **kwargs):
``` 
even though this might be the best solution at this point.

At this point, how about separating the Joint changes for everything else so we can close the others off?

Reply:
I just tried removing the arguments from the abstract methods and this seems to work:
```py
    @abstractmethod
    def connect_to(self, other: Joint):
        """All derived classes must provide a connect_to method"""
        raise NotImplementedError

    @abstractmethod
    def relative_to(self, other: Joint) -> Location:
        """Return relative location to another joint"""
        raise NotImplementedError
```
Is that all that needs to be done?

Reply:
:facepalm: yes, done.